spec.html

```diff
@@ -87,80 +48,59 @@ <h1>Uint8Array.prototype.toHex ( )</h1>
+    1. Let _strict_ be ToBoolean(? Get(_opts_, *"strict"*)).
+    1. NOTE: The order of validation and decoding in the algorithm below is not observable. Implementations are encouraged to perform them in whatever order is most efficient, possibly interleaving validation and stripping whitespace with decoding, as long as the behaviour is observably equivalent.
+    1. Let _input_ be StringToCodePoints(_string_).
+    1. If _alphabet_ is *"base64url"*, then
+      1. If _input_ contains U+002B (PLUS SIGN) or U+002F (SOLIDUS), throw a *SyntaxError* exception.
+      1. Replace all occurrences of U+002D (HYPHEN-MINUS) in _input_ with U+002B (PLUS SIGN).
+      1. Replace all occurrences of U+005F (LOW LINE) in _input_ with U+002F (SOLIDUS).
+    1. NOTE: When _strict_ is *false*, the algorithm below is equivalent to the <a href="https://infra.spec.whatwg.org/#forgiving-base64-decode">forgiving-base64 decode</a> operation in HTML.
+    1. If _strict_ is *false*, then
+      1. Remove all occurrences of U+0009 (TAB), U+000A (LF), U+000C (FF), U+000D (CR), and U+0020 (SPACE) from _input_.
+    1. Let _inputLength_ be the length of _input_.
+    1. If _inputLength_ modulo 4 is 0, then
+      1. If _input_ is not empty and the last element of _input_ is U+003D (EQUALS SIGN), then
+        1. Remove the last element of _input_.
+        1. Set _inputLength_ to _inputLength_ - 1.
+        1. If _input_ is not empty and the last element of _input_ is U+003D (EQUALS SIGN), then
+          1. Remove the last element of _input_.
+          1. Set _inputLength_ to _inputLength_ - 1.
     1. Else,
-      1. Assert: _alphabet_ is *"base64url*".
-      1. If _characters_ cannot result from applying the base64url encoding specified in section 5 of RFC 4648 to some sequence of bytes, throw a *SyntaxError* exception.
-      1. Let _bytes_ be the unique sequence of bytes such that applying the base64url encoding specified in section 5 of RFC 4648 to that sequence would produce _characters_.
-    1. Let _resultLength_ be the number of bytes in _bytes_.
-    1. Let _result_ be ? AllocateTypedArray(*"Uint8Array"*, %Uint8Array%, %Uint8Array.prototype%, _resultLength_).
+      1. If _strict_ is *true*, throw a *SyntaxError* exception.
+    1. If _input_ contains any elements which are not also elements of the standard base64 alphabet, throw a *SyntaxError* exception.
+    1. Let _lastChunkSize_ be _inputLength_ modulo 4.
+    1. If _lastChunkSize_ is 1, then
+      1. Throw a *SyntaxError* exception.
+    1. Else if _lastChunkSize_ is 2 or _lastChunkSize_ is 3, then
+      1. Append U+0041 (LATIN CAPITAL LETTER A) to _lastChunkSize_ a total of (4 - _lastChunkSize_) times.
+    1. Let _bytes_ be the unique sequence of bytes such that applying the base64 encoding specified in section 4 of RFC 4648 to that sequence would produce _input_.
```

Comment:
It's not obvious that this step sets _bytes_ to an empty sequence when _input_ is an empty list, so that's probably worth special mention or previous handling.

Reply:
Changed to "Let _bytes_ be the unique (possibly empty) sequence of bytes [...]"